src/odemis/driver/test/pwrmccdaq_test.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+'''
+Created on 21 Feb 2024
+
+@author: Stefan Sneep
+
+Copyright © 2024 Stefan Sneep, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+import copy
+import logging
+import time
+import unittest
+
+from odemis import model
+from odemis.driver import pwrmccdaq
+
+logging.getLogger().setLevel(logging.DEBUG)
+
+CONFIG_MCC = {
+                "name": "Interlock test", "role": "bsd-interlock",
+                "mcc_device": "fake",
+                "di_channels": {10: ["interlockTriggered", False]},
+              }
+
+CONFIG_LIGHT_DEVICE = {
+                "name": "Laser Hub", "role": "light",
+                "mcc_device": "fake",
+                "ao_channels": [0, 1],
+                "do_channels": [5, 6],
+                "spectra": [
+                    [592.e-9, 593.e-9, 594.e-9, 595.e-9, 596.e-9],
+                    [592.e-9, 593.e-9, 594.e-9, 595.e-9, 596.e-9]],
+                "pwr_curve": [
+                    {
+                        0: 0,
+                        5: 0.02,  # 20mW light emitter
+                    },
+                    {
+                        0: 0,
+                        5: 0.06,  # 60mW light emitter
+                    }],
+                "di_channels": {10: ["interlockTriggered", False]},
+}
+
+
+# django implementation of tag for use as a decorator in unittest (from django.test.utils)
+def tag(*tags):
+    """
+    Decorator to add tags to a test class or method.
+    """
+    def decorator(obj):
+        setattr(obj, 'tags', set(tags))
+        return obj
+    return decorator
+
+
+class TestMCCDevice(unittest.TestCase):
+    def setUp(self):
+        method = getattr(self, self._testMethodName)
+        tags = getattr(method, 'tags', {})
+        if 'skip_setup' in tags:
+            self.mcc_device = None
+            return  # setUp skipped
+
+        self.mcc_device = pwrmccdaq.MCCDevice(**CONFIG_MCC)
+        time.sleep(1)
```

Comment:
I think this whole thing could be simplified by having just a function called like `_create_device()` which would be called from each function which needs the setup.